system/Database/BaseConnection.php

```diff
@@ -593,11 +593,11 @@ public function addTableAlias(string $table)
 	/**
 	 * Executes the query against the database.
 	 *
-	 * @param string $sql
+	 * @param mixed $sql
```

Comment:
`mixed` means anything. It does not help for code reading.

Can you change it to `string|array`?

Reply:
It may be even needed to use object for such so it can be string|array|object but the main problem is type hint in the method definition. Since merging this one will break all the exiting user implementation where any of these methods have override.